vpr/src/route/route_timing.cpp

```diff
@@ -2007,35 +2006,66 @@ bool should_setup_lower_bound_connection_delays(int itry, const t_router_opts& /
 static bool is_better_quality_routing(const vtr::vector<ClusterNetId, t_traceback>& best_routing,
                                       const RoutingMetrics& best_routing_metrics,
                                       const WirelengthInfo& wirelength_info,
-                                      std::shared_ptr<const SetupHoldTimingInfo> timing_info) {
+                                      std::shared_ptr<const SetupHoldTimingInfo> timing_info,
+                                      const t_router_opts& router_opts) {
     if (best_routing.empty()) {
         return true; //First legal routing
     }
 
     //Rank first based on sWNS, followed by other timing metrics
     if (timing_info) {
-        if (timing_info->setup_worst_negative_slack() > best_routing_metrics.sWNS) {
-            return true;
-        } else if (timing_info->setup_worst_negative_slack() < best_routing_metrics.sWNS) {
-            return false;
-        }
+        if (router_opts.routing_budgets_algorithm != YOYO) {
+            // If RCV is disabled prioritize setup slack and don't worry about hold
 
-        if (timing_info->setup_total_negative_slack() > best_routing_metrics.sTNS) {
-            return true;
-        } else if (timing_info->setup_total_negative_slack() < best_routing_metrics.sTNS) {
-            return false;
-        }
+            if (timing_info->hold_worst_negative_slack() > best_routing_metrics.hWNS) {
```

Comment:
This is the RCV off case?  If so, why is it making decisions based on hold?

Reply:
I guess that check was there as a secondary quality check if the setup slacks were tied, to allow the router to pick routing with improved hold. I get that this would be unlikely though, so I guess it makes sense to remove this check for non-RCV routing.